src/apps/schedule/service/schedule.py

```diff
@@ -490,9 +500,85 @@ async def get_events_by_user(
                 PublicEventByUser(
                     applet_id=applet_id,
                     events=[
-                        PublicEvent(**event.dict()) for event in all_events
+                        self._convert_to_dto(event=event)
+                        for event in all_events
                     ],
                 )
             )
 
         return events
+
+    def _convert_to_dto(self, event: EventFull) -> ScheduleEventDto:
+        """Convert event to dto."""
+        return ScheduleEventDto(
+            id=event.id,
+            entityId=event.activity_id if event.activity_id else event.flow_id,
+            timers=TimerDto(
+                timer=HourMinute(
+                    hours=event.timer.seconds // 3600,
+                    minutes=event.timer.seconds // 60 % 60,
+                )
+                if event.timer_type == TimerType.timer
+                else None,
+                idleTimer=HourMinute(
+                    hours=event.timer.seconds // 3600,
+                    minutes=event.timer.seconds // 60 % 60,
+                )
+                if event.timer_type == TimerType.idle
+                else None,
+            ),
+            availabilityType=AvailabilityType.AlwaysAvailable
+            if event.periodicity.type == PeriodicityType.always
+            else AvailabilityType.ScheduledAccess,
+            availability=EventAvailabilityDto(
+                oneTimeCompletion=event.one_time_completion,
+                periodicityType=event.periodicity.type,
+                timeFrom=HourMinute(
+                    hours=event.start_time.hour,
+                    minutes=event.start_time.minute,
+                ),
+                timeTo=HourMinute(
+                    hours=event.end_time.hour, minutes=event.end_time.minute
+                ),
+                allowAccessBeforeFromTime=event.access_before_schedule,
+                startDate=event.periodicity.start_date,
+                endDate=event.periodicity.end_date,
+            ),
+        )
```

Comment:
Don't you think that this instance creating is overcomplicated?
As far as it is hard to read I would recommend moving out parts like this:
```python
availabilityType=AvailabilityType.AlwaysAvailable
            if event.periodicity.type == PeriodicityType.always
            else AvailabilityType.ScheduledAccess
```